pkg/addonmanager/controllers/agentdeploy/utils.go

```diff
@@ -440,3 +461,25 @@ func getManifestConfigOption(agentAddon agent.AgentAddon) []workapiv1.ManifestCo
 	}
 	return manifestConfigs
 }
+
+func getDeletionOrphaningRule(obj runtime.Object) (*workapiv1.OrphaningRule, error) {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	plural, _ := meta.UnsafeGuessKindToResource(gvk)
```

Comment:
why not return error here?

you might want to access annotation at first.

Reply:
the second output is not error,  `UnsafeGuessKindToResource` returns plural and singular.
```
func UnsafeGuessKindToResource(kind schema.GroupVersionKind) ( /*plural*/ schema.GroupVersionResource /*singular*/, schema.GroupVersionResource) {
}
```